x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobResultsProvider.java

```diff
@@ -1259,16 +1258,14 @@ private void modelSnapshots(
             size
         );
 
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder().sort(sb)
+            .query(finalQuery)
+            .from(from)
+            .size(size)
+            .trackTotalHits(true)
+            .fetchSource(REMOVE_QUANTILES_FROM_SOURCE);
         SearchRequest searchRequest = new SearchRequest(indexName);
```

Comment:
Could this fluent builder pattern be used here too?
```
        SearchRequest searchRequest = new SearchRequest(indexName)
            .indicesOptions(...)
            .source(...)
```

Reply:
No necessarily, we use the default indices options created in the request.